publication/2024-04-05_announcing_ngxs_17/article.md

```diff
@@ -0,0 +1,567 @@
+
+NGXS schematics are a new feature that makes it easier to use NGXS by automating the installation, and creation of common NGXS concepts, such as Store, Actions, and State.
+
+To auto-configure NGXS to your application, all you have to do is to execute the following command in your terminal:
+
+```bash
+ng add @ngxs/store
+```
+
+Does your workspace have multiple projects? No problem! You'll be prompted to select the project you want to add NGXS to.
+
+NGXS plugins: We couldn't resist adding [plugins](https://www.ngxs.io/plugins) to the schematics. Now you can choose which plugins you want to use, and we'll install the necessary packages and configure them for you automatically!
+
+> All the schematics support both non-standalone and standalone Angular apps.
+
+Let's see some examples:
+
+- To create a `Store` :
+
+  ```bash
+  ng generate @ngxs/store:store
+  ```
+
+  ```ts
+  import { provideStore } from '@ngxs/store';
+
+  export const appConfig: ApplicationConfig = {
+    providers: [provideStore([], { developmentMode: !environment.production })]
+  };
+  ```
+
+- To create a `State`:
+
+  ```bash
+  ng generate @ngxs/store:state --name TodosState
+  ```
+
+  ```ts
+  import { provideStore } from '@ngxs/store';
+  import { TodosState } from 'your/path';
+
+  export const appConfig: ApplicationConfig = {
+    providers: [
+      provideStore(
+        [TodosState], // <--
+        { developmentMode: !environment.production }
+      )
+    ]
+  };
+  ```
+
+- To create `Actions`:
+
+  ```bash
+  ng generate @ngxs/store:actions --name todos
+  ```
+
+  It will create the `todos.actions.ts` file
+
+---
+
+## Error Handling
+
+NGXS provides a robust error handling mechanism that automatically catches unhandled exceptions thrown within actions. These errors are then directed to a centralized `ErrorHandler`.
+
+However, you can override the default implementation and provide your custom one
+
+You can read more [here](https://www.ngxs.io/concepts/store/error-handling#custom-unhandled-error-handler)
+
+---
+
+## Bug Fixes
+
+- Show error when state initialization order is invalid [#2066](https://github.com/ngxs/store/pull/2066)
+
+  When the `UpdateState` is dispatched before the `InitState` a `console.error` will appear, and the the state initialization order must be update. An incorrect order may prevent `ngxsOnInit` from being called on feature states or lead to other unpredictable errors.
+
+- Log feature states added before store is initialized [#2067](https://github.com/ngxs/store/pull/2067)
+
+  When a feature state is initialized before the `store`, will appear an error message and the state initialization order must be updated. This typically occurs when `NgxsModule.forFeature` or `provideStates` is called before `NgxsModule.forRoot` or `provideStore`.
+
+---
+
+## Plugin Improvements
```

Comment:
I believe that plugin improvements should be placed at the end. This is because it describes that some of the symbols are now privately exported, which is less prioritized compared to other information we're providing in these docs.

Reply:
good point! I moved it just before the `NGXS Community` section. My thought was to have the `NGXS community` section as the closure.
Do you think the `Plugin Improvements` should be the last section of the readme?

Reply:
Maybe we should have the discord link at the very top, like 'first of all, we're moving to discord' or something?

Reply:
Two things about the Discord question

1. Is this "move" to Discord in relation to Slack being the default currently (from what I can tell at the moment)? I joined the Slack recently as it seemed to be pinned to the top of the `README.md`. If so, in my opinion I definitely would highlight the move to Discord towards the top of this announcement post, or at the least not at the very bottom.
2. If I am correct in this assumption, make sure to update the `README.md` badge as well as I assume the NGXS Labs contact part.

Reply:
Thanks for your comment @michael-small 
Indeed makes sense to move that on top. As of your second point, the `README.md` file has been updated and currently is part of the development version